jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/TrailingSlashAliasChecker.java

```diff
@@ -0,0 +1,39 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.server;
+
+import java.util.Objects;
+
+import org.eclipse.jetty.util.component.AbstractLifeCycle;
+import org.eclipse.jetty.util.resource.Resource;
+
+/**
+ * <p>This will approve an alias where the only difference is a trailing slash.</p>
```

Comment:
Can you provide an example in this javadoc?

Something like ... "Real resource on disk is a directory `/path/to/docroot/css/` and the resource requested was `/css` ..."

Reply:
I have added an example to this javadoc.

Reply:
I read your new javadoc.

It's not clear what is the requested path vs the `Resource` path, and what state the `Resource` path is (is it a file? is it a directory? does it exist?)

Reply:
I thought these were quite obvious given the context of it being `index.html`.

But I have tried to clarify these points in the javadoc. 
Let me know if its ok, otherwise maybe you can just rewrite it how you want and post it as a comment here.